### PR TITLE
feat: BROS-199: Enhanced paragraph annotation with auto-selection

### PR DESCRIPTION
**Pull request by cloudmark**
## BROS-199: Enhanced Paragraph Annotation with Auto-Selection and Audio Sync

This PR introduces a major upgrade to the paragraph annotation workflow, especially for dialogue-structured audio transcripts. The new features are demonstrated in this [Loom video](https://www.loom.com/share/76ff7f8655fb4a83b597a3f2195f94d4).

### **Feature Highlights**
- **Auto-unselect labels on phrase click:** Clicking a new phrase clears any selected labels to avoid confusion.
- **Auto-annotate on label selection:** When a phrase is active and a label is clicked, the entire phrase text is automatically annotated.
- **Prevent duplicate annotations:** Auto-annotation is skipped if the phrase already has existing labels/annotations.
- **Prioritize manual selection:** User text selection always takes precedence over automatic phrase annotation.
- **Auto-select created regions:** Newly created annotations are automatically selected for immediate editing.
- **Enhanced triple-click support:** Improved text selection that works properly with existing annotations.
- **Audio-phrase synchronization:** Clicking phrases sends seek messages to audio for precise playback sync.
- **Click-away to deselect:** Clicking on empty space within the paragraph area now clears any active region selections (feature-flagged).
- **Feature flag controlled:** All enhanced behaviors are gated behind `fflag_feat_front_lsdv_e_279_enhanced_paragraph_annotation_short` for safe rollout.

### **Technical Changes**
- Added MobX reaction in `ParagraphsModel` to observe label selection changes.
- Implemented `seekToPhrase` method with audio sync and label management.
- Added debounced annotation processing in `HtxParagraphs` for better UX.
- Enhanced `captureDocumentSelection` for robust triple-click handling.
- Used optional chaining for robust event handler calls (e.g., `item.seekToPhrase?.(idx)`).
- Created feature flag: `fflag_feat_front_lsdv_e_279_enhanced_paragraph_annotation_short`.
- Added comprehensive unit tests for `Phrases` component and `ParagraphsModel`, including feature-flag-aware and edge case tests.
- Added e2e test suite (`paragraphs-enhanced.test.js`) with 4 working scenarios.
- Refactored tests to be robust to UI changes and feature flag state.

### **Files Modified**
- `libs/editor/src/tags/object/Paragraphs/model.js`
- `libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx`
- `libs/editor/src/tags/object/Paragraphs/Phrases.jsx`
- `libs/editor/src/utils/feature-flags.ts`
- `libs/editor/src/tags/object/Paragraphs/__tests__/Phrases.test.tsx`
- `libs/editor/src/tags/object/Paragraphs/__tests__/model.test.ts`
- `libs/editor/tests/e2e/tests/paragraphs-enhanced.test.js`

---

### **Reason for Change**
Manual annotation of dialogue transcripts was slow and error-prone. This PR streamlines the process, enabling single-click phrase labeling, automatic region selection, and seamless audio sync, while maintaining full manual control for power users.

### **Rollout Strategy**
All new behaviors are protected by the feature flag `fflag_feat_front_lsdv_e_279_enhanced_paragraph_annotation_short` for safe, incremental rollout.

### **Testing**
- **Unit tests:** Comprehensive coverage for all new and edge-case behaviors, including feature-flag toggling.
- **E2E tests:** Four scenarios in `paragraphs-enhanced.test.js` verify the full user flow.
- **Manual QA:** See [Loom demo video](https://www.loom.com/share/76ff7f8655fb4a83b597a3f2195f94d4) for a walkthrough of the new experience.

### **Risks**
- Minimal, as all changes are feature-flagged.
- Some UI/UX changes may require user retraining once the flag is enabled by default.

### **Reviewer Notes**
- Please see the [Loom video demo](https://www.loom.com/share/76ff7f8655fb4a83b597a3f2195f94d4) for a visual walkthrough.
- All new logic is isolated and covered by tests.
- Let me know if you need any additional screenshots or test scenarios.

---

**Summary:**  
This PR significantly improves annotation efficiency and user experience for audio transcripts, while maintaining safety and test coverage.  
Demo: [Loom video](https://www.loom.com/share/76ff7f8655fb4a83b597a3f2195f94d4)

### CODE REVIEW

File: web/libs/editor/src/utils/feature-flags.ts [hunk 1]
```diff
@@ -63,6 +63,13 @@ export const FF_DEV_4174 = "fflag_fix_back_dev_4174_overlap_issue_experiments_10
  */
 export const FF_LSDV_E_278 = "fflag_feat_front_lsdv_e_278_contextual_scrolling_short";
 
+/**
+ * Enhanced paragraph annotation with automatic label selection and smart duplicate prevention
+ * Enables automatic phrase annotation when labels are selected, with priority for user text selection
+ * @link https://app.launchdarkly.com/default/production/features/fflag_feat_front_lsdv_e_279_enhanced_paragraph_annotation_short
+ */
+export const FF_LSDV_E_279 = "fflag_feat_front_lsdv_e_279_enhanced_paragraph_annotation_short";
```
Comment (nick-skriabin) [inline web/libs/editor/src/utils/feature-flags.ts]: is that expected that the feature flag is with Aha! identifier? i see the task is BROS-199, but the FF points to a completely different one

File: web/libs/editor/src/tags/object/Paragraphs/model.js [hunk 2]
```diff
@@ -403,6 +407,216 @@ const PlayableAndSyncable = types
...
+            `Skipping automatic annotation for phrase ${self.playingId}: existing labels=${hasExistingLabels}, existing results=${hasExistingResults}`,
+          );
+        }
+      }
+    },
+
+    afterCreate() {
+      // Set up reaction to observe label selection changes (enhanced annotation feature)
+      if (isFF(FF_LSDV_E_279)) {
+        self._labelReactionDisposer = reaction(
```
Comment (hlomzik) [inline web/libs/editor/src/tags/object/Paragraphs/model.js:589]: This `reaction` tracks values we are in a full control of, I suggest to use direct handlers in `*Labels`

**Review by hlomzik — CHANGES_REQUESTED** [review]

File: web/libs/editor/src/tags/object/Paragraphs/model.js [hunk 3]
```diff
@@ -403,6 +407,216 @@ const PlayableAndSyncable = types
...
+      const phrase = self._value[idx];
+      if (!isDefined(phrase.start)) return;
+
+      // If this phrase is already active, don't send seek messages (user might just be selecting text)
+      if (self.playingId === idx) {
+        return;
+      }
+
+      // Unselect regions that don't belong to the current phrase (enhanced annotation feature)
+      if (isFF(FF_LSDV_E_279)) {
```
Comment (hlomzik) [inline web/libs/editor/src/tags/object/Paragraphs/model.js]: `seekToPhrase()` is only attached with FF on, so we should either attach it always to Phrase's `onClick` or remove FF checks inside the function.

File: web/libs/editor/src/utils/feature-flags.ts (hunk 1 above)
Comment (cloudmark) [reply web/libs/editor/src/utils/feature-flags.ts]: agree, will change this.
